src/spec/create_extension_spec.py

```diff
@@ -1,47 +1,394 @@
+                                                                             'Columns are the running index. '
+                                                                             'See also https://alleninstitute.github.io/MIES/labnotebook-docs.html.',
+                                                                             neurodata_type_inc='LabNotebookTextualKeys'
+                                                                             ),
+                                                              NWBDatasetSpec(doc='Textual labnotebook values: Rows are the '
+                                                                             'running index, Columns hold the different entry names, Layers (up to nine) hold '
+                                                                             'the headstage dependent data in the first 8 and the headstage independent data in the 9th layer. '
+                                                                             'See also https://alleninstitute.github.io/MIES/labnotebook-docs.html.',
+                                                                             neurodata_type_inc='LabNotebookTextualValues'
+                                                                             )
+                                                              ]
+                                                    )
+                                       ]
+                               )
+
+    # Since 05cb3ffe (NWB: Export single test pulses as well on interactive use, 2017-11-07)
+
+    stored_testpulses = NWBDatasetSpec(doc='Raw AD testpulse data',
+                                       quantity='*',
+                                       neurodata_type_def='TestpulseRawData',
+                                       neurodata_type_inc='NWBData',
+                                       shape=(None))
+
+    # See da33596f (Rework TPStorage completely, 2018-10-18) for the switch
+    # from active AD to headstages
+    #
+    # Since 916f26e4 (TPStorage: Keep all data in one wave, 2018-10-17) we
+    # only have one TPStorage
+
+    tpstorage = NWBDatasetSpec(doc='Metadata about the Testpulse: Rows are the '
+                               'running index, Columns are active AD channels '
+                               '(up to version <= 7) or headstages (version >= 8), '
+                               'the data is in the Layers.',
+                               neurodata_type_def='TestpulseMetadata',
+                               neurodata_type_inc='NWBData',
+                               quantity='*',
+                               shape=(None, None, None))
+
+    testpulse_device = NWBGroupSpec(doc='Device for the testpulse data',
+                                    neurodata_type_inc='TestpulseDevice',
```

Comment:
I think this should be neurodata_type_def. Also, should this inherit from a type?
```suggestion
                                    neurodata_type_def='TestpulseDevice',
```